src/main/java/gift/product/domain/ProductOption.java

```diff
@@ -0,0 +1,96 @@
+package gift.product.domain;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import jakarta.persistence.UniqueConstraint;
+
+@Entity
+@Table(
+    name = "product_option",
+    uniqueConstraints = @UniqueConstraint(
+        name = "uk_product_option_product_id_name",
+        columnNames = {"product_id", "name"}
+    )
+)
+public class ProductOption {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private int quantity;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "product_id")
+    private Product product;
+
+    private static final int QUANTITY_MIN_VALUE = 1;
+    private static final int QUANTITY_MAX_VALUE = 100000000;
+
+    protected ProductOption() {
+    }
+
+    public void setProduct(Product product) {
+        this.product = product;
+    }
+
+    public void addQuantity(int quantity) {
+        int result = this.quantity + quantity;
+        if (result >= QUANTITY_MAX_VALUE) {
+            this.quantity = result;
+        } else {
+            throw new IllegalArgumentException("상품 수량 최대값(100000000) 초과입니다.");
```

Comment:
최대값이 추후 수정된다면 위의 상수와 이 에러 메시지도 같이 수정해야할 것 같은데요. 
아래와 같이 변경한다면 추후 최댓값이 변경되면 상수만 변경하면 되지 않을까요? 🤗 

```suggestion
            throw new IllegalArgumentException("상품 수량 최대값(" +  QUANTITY_MAX_VALUE + ") 초과입니다.");
```